File: _exercises/09_pdf/09_pdf.py
# %%
import chatlas
import dotenv
from pyhere import here

dotenv.load_dotenv()

# %%
recipe_pdfs = here("data/recipes/pdf/")
pdf_cheesesteak = recipe_pdfs / "PhillyCheesesteak.pdf"

# %% [markdown]
# Ask OpenAI's `gpt-4.1-nano` to turn this messy PDF print-out of a Philly
# Cheesesteak recipe into a clean list of ingredients and steps to follow.

# %%
chat = chatlas.____
chat.chat(
    "____",
    chatlas.____(pdf_cheesesteak),
)

File: _exercises/09_pdf/09_pdf.R
library(ellmer)

recipe_pdfs <- here::here("data/recipes/pdf")
pdf_waffles <- file.path(recipe_pdfs, "CinnamonPeachOatWaffles.pdf")

# Ask OpenAI's `gpt-4.1-nano` to turn this messy PDF print-out of a waffle
# recipe into a clean list of ingredients and steps to follow.
chat <- ____
chat$chat(
  "____",
  ____(pdf_waffles)
)
